### ISSUE TITLE
Crash - BadTokenException unable to add window

### ISSUE THREAD

**Issue by tuomas2** [AndBible/and-bible#2039]
https://support.andbible.org/scp/tickets.php?id=1179

```
01-01 08:45:23.466  2350  2350 E AndroidRuntime: FATAL EXCEPTION: main
01-01 08:45:23.466  2350  2350 E AndroidRuntime: Process: net.bible.android.activity, PID: 2350
01-01 08:45:23.466  2350  2350 E AndroidRuntime: android.view.WindowManager$BadTokenException: Unable to add window -- token android.os.BinderProxy@2d2c5d4 is not valid; is your activity running?
01-01 08:45:23.466  2350  2350 E AndroidRuntime: 	at android.view.ViewRootImpl.setView(ViewRootImpl.java:679)
01-01 08:45:23.466  2350  2350 E AndroidRuntime: 	at android.view.WindowManagerGlobal.addView(WindowManagerGlobal.java:342)
01-01 08:45:23.466  2350  2350 E AndroidRuntime: 	at android.view.WindowManagerImpl.addView(WindowManagerImpl.java:93)
01-01 08:45:23.466  2350  2350 E AndroidRuntime: 	at android.widget.Toast$TN.handleShow(Toast.java:467)
01-01 08:45:23.466  2350  2350 E AndroidRuntime: 	at android.widget.Toast$TN$2.handleMessage(Toast.java:350)
01-01 08:45:23.466  2350  2350 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:102)
01-01 08:45:23.466  2350  2350 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:154)
01-01 08:45:23.466  2350  2350 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6138)
01-01 08:45:23.466  2350  2350 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
01-01 08:45:23.466  2350  2350 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:893)
01-01 08:45:23.466  2350  2350 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:783)
```

**Comment by tuomas2** [comment]
We probably try to show toast when there's no window.

https://stackoverflow.com/questions/18662239/android-view-windowmanagerbadtokenexception-unable-to-add-window-on-buider-s

**Comment by tuomas2** [comment]
Was with older version, asked user if still reproduced

**Comment by tuomas2** [comment]
https://support.andbible.org/scp/tickets.php?id=1494

```
06-15 12:01:28.933 23363 23450 E AndroidRuntime: FATAL EXCEPTION: DefaultDispatcher-worker-10
06-15 12:01:28.933 23363 23450 E AndroidRuntime: Process: net.bible.android.activity, PID: 23363
06-15 12:01:28.933 23363 23450 E AndroidRuntime: android.view.WindowManager$BadTokenException: Unable to add window -- token android.os.BinderProxy@cf920c8 is not valid; is your activity running?
06-15 12:01:28.933 23363 23450 E AndroidRuntime: 	at android.view.ViewRootImpl.setView(ViewRootImpl.java:1588)
06-15 12:01:28.933 23363 23450 E AndroidRuntime: 	at android.view.WindowManagerGlobal.addView(WindowManagerGlobal.java:509)
06-15 12:01:28.933 23363 23450 E AndroidRuntime: 	at android.view.WindowManagerImpl.addView(WindowManagerImpl.java:133)
06-15 12:01:28.933 23363 23450 E AndroidRuntime: 	at android.app.Dialog.show(Dialog.java:512)
06-15 12:01:28.933 23363 23450 E AndroidRuntime: 	at android.app.AlertDialog$Builder.show(AlertDialog.java:1157)
06-15 12:01:28.933 23363 23450 E AndroidRuntime: 	at net.bible.android.view.activity.download.DownloadActivity$warnUserBooksNotDownloaded$1$1.invokeSuspend(DownloadActivity.kt:254)
06-15 12:01:28.933 23363 23450 E AndroidRuntime: 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
06-15 12:01:28.933 23363 23450 E AndroidRuntime: 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
06-15 12:01:28.933 23363 23450 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:938)
06-15 12:01:28.933 23363 23450 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:99)
06-15 12:01:28.933 23363 23450 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:226)
06-15 12:01:28.933 23363 23450 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:313)
06-15 12:01:28.933 23363 23450 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8669)
06-15 12:01:28.933 23363 23450 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
06-15 12:01:28.933 23363 23450 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
06-15 12:01:28.933 23363 23450 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)
```

**Comment by tuomas2** [comment]
Here similar stack trace, but context a bit different (crash screenshot is from welcome screen). 


https://support.andbible.org/scp/tickets.php?id=1608

```
--------- beginning of crash
09-26 03:21:32.346 25633 25745 E AndroidRuntime: FATAL EXCEPTION: DefaultDispatcher-worker-5
09-26 03:21:32.346 25633 25745 E AndroidRuntime: Process: net.bible.android.activity, PID: 25633
09-26 03:21:32.346 25633 25745 E AndroidRuntime: android.view.WindowManager$BadTokenException: Unable to add window -- token android.os.BinderProxy@76bbbba is not valid; is your activity running?
09-26 03:21:32.346 25633 25745 E AndroidRuntime: 	at android.view.ViewRootImpl.setView(ViewRootImpl.java:821)
09-26 03:21:32.346 25633 25745 E AndroidRuntime: 	at android.view.WindowManagerGlobal.addView(WindowManagerGlobal.java:369)
09-26 03:21:32.346 25633 25745 E AndroidRuntime: 	at android.view.WindowManagerImpl.addView(WindowManagerImpl.java:93)
09-26 03:21:32.346 25633 25745 E AndroidRuntime: 	at android.app.Dialog.show(Dialog.java:377)
09-26 03:21:32.346 25633 25745 E AndroidRuntime: 	at android.app.AlertDialog$Builder.show(AlertDialog.java:1121)
09-26 03:21:32.346 25633 25745 E AndroidRuntime: 	at net.bible.android.view.activity.download.DownloadActivity$warnUserBooksNotDownloaded$1$1.invokeSuspend(DownloadActivity.kt:253)
09-26 03:21:32.346 25633 25745 E AndroidRuntime: 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
09-26 03:21:32.346 25633 25745 E AndroidRuntime: 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
09-26 03:21:32.346 25633 25745 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:795)
09-26 03:21:32.346 25633 25745 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:99)
09-26 03:21:32.346 25633 25745 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:166)
09-26 03:21:32.346 25633 25745 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6861)
09-26 03:21:32.346 25633 25745 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
09-26 03:21:32.346 25633 25745 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:450)
09-26 03:21:32.346 25633 25745 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
--------- beginning of main
```

**Comment by tuomas2** [comment]
https://support.andbible.org/scp/tickets.php?id=2399

**Comment by tuomas2** [comment]
https://stackoverflow.com/questions/52649353/android-view-windowmanagerbadtokenexception-unable-to-add-window-at-toast

**Comment by tuomas2** [comment]
Fixed in d1be25f615339ec9e3f0fc0379adafcb51358ca0
